psi4/src/psi4/libpsipcm/psipcm.cc

```diff
@@ -28,416 +28,351 @@
 #ifdef USING_PCMSolver
 
 #include "psipcm.h"
-#include "psi4/libmints/matrix.h"
-#include "psi4/libmints/vector.h"
-#include "psi4/libmints/molecule.h"
+
+#include "psi4/psi4-dec.h"
+
 #include "psi4/libmints/basisset.h"
-#include "psi4/liboptions/liboptions.h"
+#include "psi4/libmints/dimension.h"
 #include "psi4/libmints/integral.h"
+#include "psi4/libmints/matrix.h"
+#include "psi4/libmints/molecule.h"
 #include "psi4/libmints/petitelist.h"
-#include <PCMSolver/PCMInput.h>
+#include "psi4/libmints/potentialint.h"
+#include "psi4/libmints/vector.h"
+#include "psi4/libpsi4util/PsiOutStream.h"
 
+#include <PCMSolver/PCMInput.h>
 
+#include <algorithm>
+#include <array>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
 
 namespace psi {
 
-void collect_atoms(std::shared_ptr<Molecule> molecule, double charges[], double centers[])
-{
+namespace detail {
+std::pair<std::vector<double>, std::vector<double>> collect_atoms(std::shared_ptr<Molecule> molecule) {
     int nat = molecule->natom();
-    for(int i = 0; i < nat; ++i) {
+    std::vector<double> charges(nat);
+    for (int i = 0; i < nat; ++i) {
         charges[i] = molecule->fZ(i);
```

Comment:
With the knowledge that `fZ(atm)` is 0. for ghosts, do any alarms go off?

Reply:
I've moved to using `Molecule::Z` instead. Great catch!